src/worker/typosquat/cache.rs

```diff
@@ -0,0 +1,100 @@
+use std::{
+    sync::{Arc, Mutex},
+    time::Instant,
+};
+
+use diesel::PgConnection;
+use typomania::{
+    checks::{Bitflips, Omitted, SwappedWords, Typos},
+    Harness,
+};
+
+use crate::swirl::PerformError;
+
+use super::{config, types::TopCrates};
+
+/// Gets the typomania harness for the cached top crates, regenerating it if it is out of date.
+pub(super) fn get_harness(
+    conn: &mut PgConnection,
+) -> Result<Arc<Harness<TopCrates>>, PerformError> {
+    HARNESS_CACHE.get(conn)
+}
+
+static HARNESS_CACHE: Cache = Cache::new();
```

Comment:
I'm not a huge fan of this since the global state makes integration tests leaky. Is there some way we can tie this to the running application/worker instance? Maybe put the cache on the `Environment` struct or something like that?

Reply:
I've moved it to `Environment`, which is kinda ugly in a different way, but does get rid of the global state.